docs/DeveloperGuide.md

```diff
@@ -308,20 +326,95 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
       Use case resumes at step 2.
 
+**Use case: Edit a person**
+
+**MSS**
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to Edit a specific person in the list
+4.  AddressBook update the person with new information
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+
+**Use case: Adding a tag to a person**
+
+**MSS**
+
+1.  User find a person
+2.  AddressBook shows the searching result of the person
+3.  User requests to add a tap to a specific person
+4.  AddressBook adds that tag to the person
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. Cannot find the user.
+
+  Use case ends.
+
+* 3a. The tag is invalid.
+
+    * 3a1. The tag exist already.
```

Comment:
I think these are supposed to be steps taken by the user/actions taken by the program instead of what may have caused the exception?

Reply:
Instead of listing out the potential causes of the extension, shouldn't these be separated out (via 3a and 3b)? 3a1 and 3a2 should be actions the program or user does.